### DISCUSSION TITLE
自定义组件x-component-props如何传递值为组件的属性了？

### DISCUSSION THREAD

**Discussion by YocoZhang** [alibaba/formily#1944] [Q&A 2.x]
![截屏2021-08-03 下午7 53 14](https://user-images.githubusercontent.com/15158322/128012256-1556175b-fa15-4a43-9830-b10ae0e211e4.png)

如children支持传递组件，目前好像是不支持组件（React element）数据格式？

**Comment by janryWang** [comment]
给个完整的codesandbox，否则无法给你帮助

**Reply by YocoZhang** [reply]
// schema
const schema = {
  "type": "object",
  "properties": {
    "formToast": {
      "type": "object",
      "x-component": "mega-layout",
      "x-component-props": {
        "full": true,
        "labelAlign": "top"
      },
      "properties": {
        "firstLine": {
          "key": "firstLine",
          "name": "firstLine",
          "x-component": "collapse",
          "x-component-props": {
            "x-content": '<Demo />',
            "title": "第一行信息"
          },
          'x-content': 'xxxx'
        }
      }
    }
  }
}


// 自定义组件

import React from 'react';
import { Collapse } from 'antd';
import { Icon } from '@dol-design/rn';
import './index.less';

const { Panel } = Collapse;

export default function Index(props) {
  const { value, title, onChange, children } = props;
  return (
    <div className="dolphin-base-collapse-warp">
      <Collapse
       ghost
       expandIconPosition="right"
       className="dolphin-base-collapse"
      >
        <Panel header={title}>
          {children}
        </Panel>
      </Collapse>
    </div>
  );
}

**Reply by YocoZhang** [reply]
不好贴代码，这个涉及工程内容比较多，贴了一些相关代码不知道可以辅助理解不？

**Reply by janryWang** [reply]
如果你是纯前端维护schema，直接children传react dom就行了，如果是后端下发的json schema，那就得用表达式，作用域传富文本构造函数，比如`scope={{text:(str)=><div>{str}</div>}}` 在使用的时候children:"{{text('hello')}}"

**Reply by YocoZhang** [reply]
是前端维护的scheme，但是直接传react dom报错了，错误信息如下，应该是不支持传react dom
![截屏2021-08-03 下午9 11 59](https://user-images.githubusercontent.com/15158322/128021417-23b0ec9a-b24c-49a0-87cc-2b3c2bac621a.png)

**Reply by janryWang** [reply]
你这个报错好奇怪，看不出来，建议你抽出个codesandbox来看看

**Comment by atzcl** [comment]
没有限制，都可以传的，`children` 做一下映射即可， https://codesandbox.io/s/mutable-thunder-sc66q?file=/App.tsx

**Reply by YocoZhang** [reply]
我目前的实现是通过SchemaForm组件传递的数据，这个怎么映射 children了？
`
<SchemaForm
          key={params.type}
          components={components}
          value={params.props}
          onChange={changeHandle}
          schema={params.formSchema}
        ></SchemaForm>`

**Reply by YocoZhang** [reply]
import { SchemaForm } from '@formily/antd';
